Receiver_C/For_Testing/Test_case.cpp

```diff
@@ -0,0 +1,169 @@
+#define CATCH_CONFIG_MAIN  // This tells Catch to provide a main() - only do this in one cpp file
+#include "catch.hpp"
+
+#include "Test_case.h"
+#include "../Common_Header.h"
+#include "../Data_Reader.h"
+#include "../Data_Analyzer.h"
+#include "../Output_Handler.h"
+#include "../Receiver_Main.h"
+
+TEST_CASE("Positive_case: Calculate SMA on given array") 
+{
+	double derived_SMA , data[5] = {1.0,3.0,5.0,7.0,9.0};	
+	double expected_SMA = 5.000000;
+    int retVal ; 	
+	retVal = Calculate_SMA( data , (sizeof(data)/(sizeof(data[0]))) , &derived_SMA );
+	REQUIRE( (retVal == OK) );
+	REQUIRE((derived_SMA == expected_SMA));		  

```

Comment:
Nice and specific tests! But always use a tolerance 'epsilon' in floating point equality